src/controllers/import.js

```diff
@@ -125,6 +129,37 @@ function ImportController(context) {
     }
   }
 
+  /**
+   * Extract the domain from a URL.
+   * @param {string} inputUrl the URL to extract the domain from.
+   * @return {string} the domain extracted from the URL.
+   */
+  function getDomain(inputUrl) {
+    const parsedUrl = new URL(inputUrl);
+    const parsedDomain = psl.parse(parsedUrl.hostname);
+    return parsedDomain.domain; // Extracts the full domain (e.g., example.co.uk)
+  }
+
+  /**
+   * Check if the URLs in urlList belong to any of the base domains.
+   * @param {string[]} urlList the list of URLs to check.
+   * @param {string[]} baseDomainList the list of base domains to check against.
+   * @return {true} if all URLs belong to an allowed base domain
+   * @throws {ErrorWithStatusCode} if any URL does not belong to an allowed base domain
+   */
+  function isUrlInBaseDomains(urlList, baseDomainList) {
+    const invalidUrls = urlList.filter((inputUrl) => {
+      const urlDomain = getDomain(inputUrl);
+      return !baseDomainList.some((baseDomain) => urlDomain === baseDomain);
+    });
+
+    if (invalidUrls.length > 0) {
+      throw new ErrorWithStatusCode(`Invalid request: URLs not allowed: ${invalidUrls.join(', ')}`, STATUS_BAD_REQUEST);
```

Comment:
This could lead to a very large `x-error` header, which might lead to other issues. Perhaps just include the first invalid URL, as an example?

Reply:
I added this change after initial reviews - we wanted to prevent back and forth with the customer if there are multiple URLs which are invalid. But, I could change it to the first URL.